apps/communications/dialer/test/marionette/lib/views/keypad/views.js

```diff
@@ -0,0 +1,54 @@
+'use strict';
+/* global module */
+var KeypadAccessor = require('./accessors');
+
+function KeypadView(client) {
+  
+  this.client = client;
+  this.accessors = new KeypadAccessor(client);
+
+}
+
+KeypadView.prototype = {
+ 
+  key: function(keyNumber) {
+    return this.client.helper.waitForElement
+      (this._getKeypadKeySelector(keyNumber));
+  },
+  _getKeypadKeySelector: function(key) {
+    return '.keypad-key[data-value="' + key +'"]';
+  },
+  
+  typePhoneNumber: function(number) {
+    var i;
+    var typeNumberArray = String(number).split('');
+
+    for (i in typeNumberArray){
+      if(i == '+'){
+        var zero_key = parseInt(typeNumberArray[i]);
+        var zero = this.key[zero_key];
+        this.actions.longPress(zero, 1).perform();
+      }
+      else{
+        var key_tapped = parseInt(typeNumberArray[i]);
+        var other_key = this.key(key_tapped);
+        other_key.tap();
+      }
+    }
+    var keypadView = new KeypadView(this.client);
```

Comment:
Tried to remove that and just to return, but the the test fails..tried several other options..but that is the only way I get it working :S